src/main/jbake/content/blog/2017/09/agent-to-services-over-ssl.adoc

```diff
@@ -0,0 +1,146 @@
+= Connecting Hawkular Agent to Hawkular Services over SSL
+Josejulio Martínez
+2017-09-11
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, services, ssl
+
+SSL provides identification of the communicating parties, confidentiality and integrity of the information being shared.
+
+In production environments, network eavesdropping could be a concern. You can use SSL to provide a secure communication
+channel between servers.
+
+Hawkular Services expects Hawkular Agents to connect, push information and listen for commands.
+These commands also include server credentials that could be vulnerable to a man-in-the-middle attack.
+
+A previous link:http://www.hawkular.org/hawkular-services/docs/installation-guide/secure-comm.html[article] shows how
+to configure your Hawkular Services with SSL and have Hawkular Agents connect to it trough SSL.
+This guide will show how is done for Dockerized Hawkular Services Agents.
+
+== Preparing your certificates
+
+Before starting we need to prepare the certificates that we are going to use.
+Your public and private key for Hawkular Services need to be on PEM or PKC12 format.
+For this guide we will use PEM.
+
+We can create self-signed certificates (in PEM format) using:
+```
+keytool -genkey -keystore hawkular.keystore -alias hawkular -dname "CN=hawkular-services" -keyalg RSA -storepass hawkular -keypass hawkular -validity 36500 -ext san=ip:127.0.0.1,dns:hawkular-services
+keytool -importkeystore -srckeystore hawkular.keystore  -destkeystore hawkular.p12 -deststoretype PKCS12 -srcalias hawkular -deststorepass hawkular -destkeypass hawkular -srcstorepass hawkular
+openssl pkcs12 -in hawkular.p12 -password pass:hawkular -nokeys -out hawkular-services-public.pem
+openssl pkcs12 -in hawkular.p12 -password pass:hawkular -nodes -nocerts -out hawkular-services-private.key
+```
+
+IMPORTANT: When creating the certificates, remember to include the host that Hawkular will use
+(as Common Name(CN) and Subject Alternative Name(SAN)), else the certificate validation will fail.
+Through this guide, host will be set to hawkular-services.
+
+WARNING: Always include san=ip:127.0.0.1 in your certificate, as this will be used internally by Hawkular Services.
+
+== Starting Cassandra
+
+Hawkular Services requires a cassandra instance, you can start one by doing:
+
+```
+docker run --name hawkular-cassandra -e CASSANDRA_START_RPC=true -d cassandra:3.0.9
```

Comment:
We recommend C* 3.0.12: 
http://www.hawkular.org/hawkular-services/docs/installation-guide/